### PR TITLE
Refactor IDs

### PR DESCRIPTION
**Pull request by ManuelLR**
**Description of the change**

I refactored the IDs in the next way:

| Original | New |
|------|--------|
| environmentCustomerPackageValues | valuesDerivedFromCartoPlatformEnvironment |
| kotsEncryptionSecretKey | autogeneratedEncryptionSecretKey |
| kotsVarnishDebugSecret | autogeneratedVarnishDebugSecret |
| kotsJwtApiSecret | autogeneratedJwtApiSecret |
| kotsVarnishPurgeSecret | autogeneratedVarnishPurgeSecret |
| projectId | derivedSelfHostedGCPProjectId |
| cartoSelfhostedDomain | cartoSelfHostedDomain |
| accessToCartoModeK8sDefaultLBLabel | accessToCartoModeK8sDefaultLabel |
| accessToCartoModeK8sDefaultLB | accessToCartoModeK8sDefault |
| accessToCartoTLSCertificatesGKE | accessToCartoTLSGKECertificateBehaviour |
| tls_managed_certificates | GCP_externally_managed_certificates |
| tls_custom_certificates | use_custom_certificates_files_provided |
| accessToCartoManagedCertsGCPName | accessToCartoGCPExternallyManagedCertsName |
| accessToCartoTLSCertificatesEKS | accessToCartoAWSExternallyManagedCertsName |
| defaultLBStaticIPAddressGCP | accessToCartoGCPStaticIPName |
| defaultLBStaticIPAddressAWS | accessToCartoAWSStaticIPName |
| externalRedisSSLMode | externalRedisTLSBehaviour |
| ssl_mode_disable (redis) | tls_disabled (redis) |
| ssl_mode_enable (redis) | tls_enabled (redis) |
| externalRedisSSLCertificate | externalRedisTLSCertificate |
| locationDataServicesGeocodingProvider | LDSGeocodingProvider |

**Benefits**

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

### CODE REVIEW

**Comment by ManuelLR** [comment]
@moicalcob I had doubts with `kubernetesGatewayEnabled` what do you think? :thinking:

File: manifests/kots-config.yaml [hunk 1]
```diff
@@ -835,8 +836,8 @@ spec:
           when: '{{repl (ConfigOptionEquals "enableGoogleWorkloadIdentity" "0") }}'
           help_text: |-
             This is the Service Account that is going to be used by the Self-Hosted environment to access CARTO platform services.
-            Please do not modify it at least that you know what you are doing.
-            Contact [Carto Support](https://docs.carto.com/carto-self-hosted/support) if you have any doubts.
+            ⚠️ Please do not modify it at least that you know what you are doing.
+            Contact [Carto Support](https://docs.carto.com/carto-self-hosted/support) before proceed and if you have any doubts.
```
Comment (moicalcob) [inline manifests/kots-config.yaml:840]:
```suggestion
            Contact [Carto Support](https://docs.carto.com/carto-self-hosted/support) before proceeding or if you have any doubts.
```

File: manifests/kots-config.yaml [hunk 2]
```diff
@@ -55,37 +55,38 @@ spec:
...
-        - name: internalRedisPassword
+        - name: autogeneratedInternalRedisPassword
           title: password used for the internal redis
           type: textarea
           hidden: true
           readonly: false
           value: '{{repl RandomString 16 "[A-Za-z0-9]" }}'
-        - name: projectId
+
+        - name: derivedSelfHostedGCPProjectId
```
Comment (moicalcob) [inline manifests/kots-config.yaml:89]: Why do we include the derived prefix in some properties? 🤔 It sounds to me as if they were "computed" values. Values obtained from other ones, no?

File: manifests/kots-config.yaml [hunk 3]
```diff
@@ -516,21 +517,21 @@ spec:
...
               title: Enable
-        - name: externalRedisSSLCertificate
-          title: SSL certificate
-          when: '{{repl (ConfigOptionEquals "externalRedisEnabled" "1") }}'
+        - name: externalRedisTLSCertificate
+          title: SSL/TLS certificate
+          when: '{{repl and (ConfigOptionEquals "externalRedisEnabled" "1") (ConfigOptionEquals "externalRedisTLSBehaviour" "tls_enabled") }}'
           help_text: |-
-            If you don't know what this is, please change the SSL Mode.
+            If you don't know what this is, please change the SSL/TLS Mode.
```
Comment (moicalcob) [inline manifests/kots-config.yaml:534]:
```suggestion
            If you don't know what this is, please disable the SSL/TLS Mode.
```

File: manifests/kots-config.yaml [hunk 4]
```diff
@@ -835,8 +836,8 @@ spec:
           when: '{{repl (ConfigOptionEquals "enableGoogleWorkloadIdentity" "0") }}'
           help_text: |-
             This is the Service Account that is going to be used by the Self-Hosted environment to access CARTO platform services.
-            Please do not modify it at least that you know what you are doing.
-            Contact [Carto Support](https://docs.carto.com/carto-self-hosted/support) if you have any doubts.
+            ⚠️ Please do not modify it at least that you know what you are doing.
```
Comment (moicalcob) [inline manifests/kots-config.yaml:839]:
```suggestion
            ⚠️ Please do not modify it unless you know what you are doing.
```

**Comment by moicalcob** [comment]
> @moicalcob I had doubts with `kubernetesGatewayEnabled` what do you think? 🤔
At least I would change it for `kubernetesGatewayAPIUsageEnabled` 🤔 IMHO it clarifies that it's for the usage of that kubernetes API

**Comment by ManuelLR** [comment]
@moicalcob In the end I think I will postpone the renaming of `kubernetesGatewayEnabled` because I need to thoroughly review that section so there is no point in doing it here.

**Review by moicalcob — APPROVED** [review]

File: manifests/kots-config.yaml (hunk 2 above)
Comment (ManuelLR) [reply manifests/kots-config.yaml:89]: Finally it was a misunderstanding and the current name was great.

**Pull request merged**